audio-worklet/src/noisegate-demo.js

```diff
@@ -36,12 +35,33 @@ class NoisegateDemo {
     this.releaseMin_ = 0;
     this.releaseMax_ = 0.1;
 
+    // There are two sound sources (speech and noise), which are mixed
+    // together and summed. From this summing junction, the signal splits into
+    // three paths, one going through the script processor noise gate, one going
+    // through the audio worklet noise gate and one bypassing the noise gate.
+    this.noiseGain_ = new GainNode(this.context_, {gain: 0});
+    this.speechAndNoiseSummingJunction_ = new GainNode(this.context_);
+    this.bypassNoisegateRoute_ = new GainNode(this.context_, {gain: 0});
+    this.activeNoisegateRoute_ = new GainNode(this.context_, {gain: 1});
+    this.masterGain_ = new GainNode(this.context_, {gain: 0.5});
+
     // The recorded speech sample is louder than -40db and not muted by default.
     this.threshold_ = -40;
     this.thresholdMin_ = -100;
     this.thresholdMax_ = 0;
-
-    // TODO: Implement audio worklet logic.
+    
+    if (this.workletIsAvailable_) {
+      this.noisegateAudioWorklet_ =
+          new AudioWorkletNode(this.context_, 'noisegate-audio-worklet');
+
+      // The script processor is used by default, and if workletGain_.gain is 0,
+      // then scriptProcessorGain_.gain is 1, and vice versa.
+      this.workletGain_ = new GainNode(this.context_, {gain: 0});
+      this.speechAndNoiseSummingJunction_.connect(
```

Comment:
It got reformatted with Clang. I have made sure the rest of the document is consistent with Clang now.